chain/epoch_manager/src/reward_calculator.rs

```diff
@@ -89,19 +89,28 @@ impl RewardCalculator {
         let total_stake: Balance = validator_stake.values().sum();
         for (account_id, stats) in validator_block_chunk_stats {
             // Uptime is an average of block produced / expected and chunk produced / expected.
-            let average_produced_numer = U256::from(
-                stats.block_stats.produced * stats.chunk_stats.expected
-                    + stats.chunk_stats.produced * stats.block_stats.expected,
-            );
-            let average_produced_denom =
-                U256::from(2 * stats.chunk_stats.expected * stats.block_stats.expected);
+            let (average_produced_numer, average_produced_denom) =
+                if stats.block_stats.expected == 0 && stats.chunk_stats.expected == 0 {
+                    (U256::from(0), U256::from(1))
+                } else if stats.block_stats.expected == 0 {
+                    (U256::from(stats.chunk_stats.produced), U256::from(stats.chunk_stats.expected))
```

Comment:
We should check that we don't break protocol..

Could we quickly iterate over all epoch stats, and check that block_stats.expected ==0 iff chunk_stats.expected == 0 ?

Reply:
Unfortunately this is actually not easy because validator stats is not stored in storage. It's not part of EpochInfo and we only store the most recent EpochInfoAggregator. So if we want to check this for all epochs, we literally need to recalculate these stats for all blocks, which will probably take a long time. I wrote a binary for that, but still need some time to run it against an archival node. To merge this PR fast, I'll just guard the code under the ChunkOnly feature.